Godeps/_workspace/src/k8s.io/kubernetes/pkg/registry/replicaset/proxy/proxy.go

```diff
@@ -0,0 +1,205 @@
+	if err != nil {
+		return nil, err
+	}
+	return &kapi.ReplicationControler{
+		ObjectMeta: rs.ObjectMeta,
+		Spec: extensions.ReplicaSetSpec{
+			Replicas: rc.Spec.Replicas,
+			Selector: selector.(map[string]string),
+			Template: rc.Spec.Template,
+		},
+		Status: extensions.ReplicaSetStatus{
+			Replicas:             rs.Status.Replicas,
+			FullyLabeledReplicas: rs.Status.FullyLabeledReplicas,
+			ObservedGeneration:   rs.Status.ObservedGeneration,
+		},
+	}
+}
+
+// convertReplicaSetList transforms a ReplicaSetList into a ReplicationControllerList.
+func convertReplicaSetList(rsList extensions.ReplicaSetList) (*kapi.ReplicationControllerList, error) {
+	rcList := &kapi.ReplicationControllerList{}
+	errs := []error{}
+	for _, rs := range rsList.Items {
+		rc, err := convertReplicaSet(rs)
+		if err != nil {
+			errs = append(errs, err)
+		} else {
+			rcList.Items = append(rcList.Items, *rc)
+		}
+	}
+	return rcList, errors.NewAggregate(errs)
+}
+
+// convertReplicationController transforms a ReplicationController into a ReplicaSet.
+func convertReplicationController(rc kapi.ReplicationControler) *extensions.ReplicaSet {
+	return &extensions.ReplicaSet{
+		ObjectMeta: rc.ObjectMeta,
+		Spec: extensions.ReplicaSetSpec{
+			Replicas: rc.Spec.Replicas,
+			Selector: unversioned.SetAsLabelSelector(labels.Set(rc.Spec.Selector)),
+			Template: rc.Spec.Template,
+		},
+		Status: extensions.ReplicaSetStatus{
+			Replicas:             rc.Status.Replicas,
+			FullyLabeledReplicas: rc.Status.FullyLabeledReplicas,
+			ObservedGeneration:   rc.Status.ObservedGeneration,
+		},
+	}
+}
+
+// convertReplicationControllerList transforms a ReplicationControllerList into a ReplicaSetList.
+func convertReplicationControllerList(rcList kapi.ReplicationControllerList) *extensions.ReplicaSetList {
+	rsList := &extensions.ReplicaSetList{}
+	for _, rc := range rcList.Items {
+		rsList.Items = append(rsList.Items, *convertReplicationController(rc))
+	}
+	return rsList
+}
+
+// NewProxyREST returns a RESTStorage object that will treat replica sets as replication controllers.
+func NewProxyREST(store rsregistry.REST, client unversionedcore.ReplicationControllersGetter) *ProxyREST {
+	return &ProxyREST{
+		store:          store,
+		client:         client,
+		createStrategy: rcregistry.Strategy,
+		updateStrategy: rcregistry.Strategy,
+	}
+}
+
+type ProxyREST struct {
+	// store works with replica sets
+	store rsregistry.REST
+	// client works with replication controllers
+	client unversionedcore.ReplicationControllersGetter
+	// allows validation of replication controllers during creation
+	createStrategy rest.RESTCreateStrategy
+	// allows validation of replication controllers during updates
+	updateStrategy rest.RESTUpdateStrategy
+}
+
+func (s *ProxyREST) New() runtime.Object   { return &extensions.ReplicaSet{} }
+func (*ProxyREST) NewList() runtime.Object { return &extensions.ReplicaSetList{} }
+
+// TODO: How do we know if we should query the rs or the rc storage?
```

Comment:
>  It can just have two object kinds under the same key.

Not sure how I can set this up. Is it something I need to configure in the generic.Etcd struct?